packages/infrastructure/src/public/saleor/search/infrastructure/search-infra.ts

```diff
@@ -25,37 +35,43 @@ export const saleorSearchInfra =
     // TODO: Find a better way how to handle filters between providers
     const attributesFilter: AttributeInput[] | undefined = filters
       ? Object.entries(filters).map(([slug, value]) => ({
-          slug: slug,
+          slug,
           boolean: value === "true" ? true : undefined,
           values: value !== "true" ? [value] : undefined,
         }))
       : undefined;
 
-    const whereFilter = {
-      ...(productIds && { ids: productIds }),
-      ...(category && { category: { oneOf: [category] } }),
-    };
-
     try {
       loggingService.debug("Fetching the products from Saleor", {
         query,
         channel: context.channel,
+        category,
+        collection,
       });
 
       const { data, error } = await graphqlClient(apiURL).execute(
         SearchProductQueryDocument,
         {
           variables: {
-            where: Object.keys(whereFilter).length ? whereFilter : undefined,
-            search: query,
+            after,
+            before,
+            categorySlug: category || null,
```

Comment:
```suggestion
            categorySlug: category ?? null,
```

and same with `collectionSlug`